primer-rel8/src/Primer/Database/Rel8/Rel8Db.hs

```diff
@@ -395,6 +432,20 @@ sessionById :: UUID -> Query (Schema.SessionRow Expr)
 sessionById sid =
   allSessions >>= filter \p -> Schema.uuid p ==. litExpr sid
 
+-- Select all sessions whose name contains the given substring.
+sessionByNameSubstr :: Text -> Query (Schema.SessionRow Expr)
+sessionByNameSubstr substr =
+  -- N.B. the arguments to 'like' are reversed from what you might expect.
+  filter (like (litExpr ("%" <> escape substr <> "%")) . Schema.name) =<< allSessions
+  where
+    -- Escape @%@ and @_@ characters in the given string, as these are
+    -- wildcards in SQL. Note that the backslash is the default escape
+    -- character, and therefore must also be escaped. Make sure we do
+    -- backslash first, as otherwise we'll double-escape the other two
+    -- replacements!
```

Comment:
~Have you tested this with both the postgresql and sqlite dbs?~ Oh, I see: this code is only for the postgresql db.